### DISCUSSION TITLE
H112-370 still supported?

### DISCUSSION THREAD

**Discussion by stellarpower** [Salamek/huawei-lte-api#229] [General]
Hi,

Thanks for this, looks really useful! I'm getting good speeds after a restart but then it gets sluggish easily, so should help with diagnosing if it's the router, physical location, or my provider.

I am getting `LoginErrorUsernamePasswordWrongException: 108006: Username and Password wrong` when trying to access my H112-370. The login only asks for password and looking at the javascript it sends, the username should indeed be sent as admin. Firmware version info from the web dashboard:

```
Hardware version             WL1H112M
Software version             10.0.3.1(H330SP51C21)
Web UI version               WEBUI 10.0.3.1(W2SP51C21)
Configuration file version   H112-370-CUST 8.0.1.3(C21)
```

Unfortunately, it's one of the ones not just locked to the network's SIM but with slightly modified firmware that hides a lot of things. Had I known the extent might have bought something else, but there we go.

In my browser I usually force HTTPS and ignore the self-signed cert, as if I use unencrypted it ties to redirect to hirouter.com which then doesn't resolve. So I modified the device_info example slightly to disable checking of the certificate.

From the web inspector it seems the login flow is a bit complicated, with nonces going back and forth, do you happen to know if there are known issues, or should this model and version still be supported?

Thanks!

**Comment by pvswie** [comment]
Hi Stellarpower,

I am also new here and have a B636-336. My device also does not have a user name when logging in, just a password. 
Thanks for the confirmation that I should use 'ADMIN' as user. Where / how did you find this?

So you got yourself a H112-370 that is highly limited in functionality. I bought a model that was also locked to a provider with limited functionality. Luckily I found out in time and was able to return it. What I ran into (but did not saveguard the URL, I was able to return my router) is an option, despite the firmware being locked, to upgrade to the regular Huawei firmware. Maybe you want to give that a try, I never tried, I do not know whether such an update would be successful.

Regarding your 'gets sluggish': I am pretty happy with the B636-336. The device is quite reliable and not getting sluggish, my provider however causes the cellular link to disconnect every 24 hours. I can also confirm that one of the examples (reconnect_dialup.py) works perfectly for my router, the router gets disconnected for just a few seconds, maybe you can run this script every now and then to get rid of the 'gets sluggish'. In addition please be aware deep down in a menu there is logging: Advanced --> System Settings --> System Logs, maybe there is something to see in those logs. 

Kind Regards
